src/Models/BaseElement.php

```diff
@@ -285,17 +285,25 @@ public function onBeforeWrite()
     {
         parent::onBeforeWrite();
 
-        if (!$this->Sort) {
-            if ($this->hasExtension(Versioned::class)) {
-                $records = Versioned::get_by_stage(BaseElement::class, Versioned::DRAFT);
-            } else {
-                $records = BaseElement::get();
-            }
+        // If a Sort has already been set, then we can exit early
+        if ($this->Sort) {
+            return;
+        }
 
-            $records = $records->filter('ParentID', $this->ParentID);
+        // If no ParentID is currently set for the Element, then we don't want to define an initial Sort yet
+        if (!$this->ParentID) {
+            return;
+        }
```

Comment:
The guts of the change.

Rationale detailed here:
https://github.com/silverstripe/silverstripe-elemental/issues/849#issuecomment-922536928